### PR TITLE
Dining token bug fix and analytics completed

### PR DESCRIPTION
**Pull request by aantenberg**
Dining analytics page finally added! Also, the dining token expiration bug is fixed (hopefully 😅)
simulator_screenshot_54C98147-CC30-4E60-B3C5-E684CC98CDE6

### CODE REVIEW

File: PennMobile/Dining/Model/PastDiningBalances.swift [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+//
+//  Created by Andrew Antenberg on 3/25/22.
+//  Copyright © 2022 PennLabs. All rights reserved.
+//
+
+import Foundation
+
+struct PastDiningBalances: Codable {
+    let balanceList: [DiningBalance]
+}
```
Comment (01jongmin) [inline PennMobile/Dining/Model/PastDiningBalances.swift:13]: I don't think it is necessary to create a new struct just to be a wrapper for an array of another struct.

File: PennMobile/Dining/SwiftUI/Views/DiningViewHeader.swift [hunk 2]
```diff
@@ -24,7 +25,8 @@ struct DiningViewHeader: View {
             }
             HStack {
                 DiningBalanceView(description: "Guest Swipes", image: Image(systemName: "person.2.fill"), balance: Double(diningVM.diningBalance.guestVisits), specifier: "%.0f")
-                AnalyticsCardView(text: "Coming Soon!")
+                AnalyticsCardView(text: "Analytics!")
```
Comment (01jongmin) [inline PennMobile/Dining/SwiftUI/Views/DiningViewHeader.swift:28]: Maybe removing the exclamation mark for consistency 😅

**Review by 01jongmin — CHANGES_REQUESTED** [review]
Just 2 minor fixes and it LGTM! 🔥

File: PennMobile/Dining/SwiftUI/Views/DiningViewHeader.swift (hunk 2 above)
Comment (aantenberg) [reply PennMobile/Dining/SwiftUI/Views/DiningViewHeader.swift:28]: Sounds good!

File: PennMobile/Dining/Model/PastDiningBalances.swift (hunk 1 above)
Comment (01jongmin) [reply PennMobile/Dining/Model/PastDiningBalances.swift:13]: When decoding, you can use `JSONDecoder().decode([DiningBalance].self, from: {})`

File: PennMobile/Dining/Model/PastDiningBalances.swift [hunk 3]
```diff
@@ -6,8 +6,8 @@
 //  Copyright © 2022 PennLabs. All rights reserved.
 //
 
-import Foundation
-
-struct PastDiningBalances: Codable {
-    let balanceList: [DiningBalance]
-}
+//import Foundation
```
Comment (aantenberg) [inline PennMobile/Dining/Model/PastDiningBalances.swift:9]: Didn't realize this was just commented out, I'll delete it and squash commits

File: PennMobile/Dining/Model/PastDiningBalances.swift (hunk 1 above)
Comment (aantenberg) [reply PennMobile/Dining/Model/PastDiningBalances.swift:13]: Resolved: it is required due to the API response type

**Review by 01jongmin — APPROVED** [review]
LGTM!

**Pull request merged**